userguide/aws/how-tos/add-a-certificate.adoc

```diff
@@ -0,0 +1,157 @@
+
+Increasingly, modern browsers require valid certificates when accessing sites using the HTTPS protocol. It may be difficult or impossible to access these sites if a valid certificate is not found.
+
+The structural details of certificates are defined by international standards and are well beyond the scope of this discussion. For more details see the https://en.wikipedia.org/wiki/X.509[Wikipedia article]. One thing you need to understand, though, is the certificate authority. Normally this is either an organization that is intrinsically well-known and trusted, such as DigiCert or GlobalSign (called a *root* CA), or an organization in the *trust hierarchy* derived from a root. A CA in a trust hierarchy must be itself certified by another CA in a chain which eventually leads to the root CA. Your CA, in turn, must be able to certify that *you* are who you say you are!
+
+For temporary use, you can sometimes create a *self-signed* certificate, in which you are your own CA. Most browsers will frown on this, since you are simply vouching for yourself, but it may allow you to get started with security. We will not consider this option in this document.
+
+If you own a domain, your domain provider can serve as your CA. If you don't have a domain, you can purchase one for very low cost from a domain provider such as Godaddy. When you obtain a domain you need to prove your identity, so your domain provider can trust you. To demonstrate this trust you will need a CNAME.
+
+== Preliminary concepts
+
+=== CNAMEs
+
+A CNAME, or *Canonical Name*, is an identifier that links your CA to the resources that you are protecting with your certificate. You will be protecting public access to your instances of Kaui and Kill Bill through your load balancer, identified by its DNS name. The CNAME will link your domain to the load balancer. If your domain is `mydomain.com` your CNAME might look like `kaui.mydomain.com`. This will direct browsers to your verified domain, and then, through your CNAME, to your AWS resources.
+
+The CNAME is part of your domain's DNS profile, and your domain provider should have instructions for setting it up. These instructions should ask you to enter two strings called `Name` and `Value`. These strings will be obtained during the certificate generation process.
+
+=== Protocol and Port Setup
+
+Your AWS security group by default provides rules giving access to Kaui and Kill Bill using the HTTP protocol and a specified port number. For HTTPS you will need additional rules with the new protocol and a new port number.
+
+=== Validation
+
+When you create a new certificate you will need to validate your certificate. This identifies the trust hierarchy for the certificate. You may need to setup a temporary CNAME for this purpose, and you may need to provide temporary access via an extra port in your security group.
+
+== Obtaining a Certificate
+
```

Comment:
Add this from Overview: "HTTPS is based on the use of a verification document called a *certificate*. the certificate serves two purposes:

1. It provides the keys necessary for a highly secure encryption.
2. It provides the electronic signature of a *certificate authority (CA)* who guarantees the identity of the sender."